core/postsolve.gms

```diff
@@ -23,7 +23,7 @@ $endif.neg
 );
 
 if(cm_iterative_target_adj eq 4,
-*JeS* Update tax levels/ multigasbudget values to reach the CO2 FF&I budget (s_actualbudgetco2 runs from 2020-2100)
+*JeS* Update tax levels / cumulative emission level to reach the FF&I budget target (s_actualbudgetco2 runs from 2020-2100, not peak budget)
```

Comment:
I understand the idea of having different words for the current budget (=cumulative emission level) and the target, but it doesn't make sense to me to "update cumulative emission levels" as we can only update the target. That's why I suggested tax levels / budgets. Also, I would keep the "CO2" in "CO2 FF&I budget"

Reply:
okay, then I think I don't quite understand how this works...  as I understand, the algorithm raises the in-iteration CO2 tax with some function, and then with this new CO2 price, the model finds an optimum and in postsolve, we can calculate the global cumulative emission level from the energy system under the new CO2 price trajectories (for all regions), and the price traj. can be regionally differentiated. And then the algo compares this cumulative level to the target, which I call the budget (like a household budget, which is fixed, as I assume my income is fixed, so the budget is not something I can adjust(?) at least intuitively I understand this as such). And then depends on the difference between cum. emission level and the target, the algo changes the CO2 price traj. again, and so on

is this roughly how it works? or do we update the budget? i.e if we put c_budgetCO2from2020 = 500, we do not adjust this number, right?

on CO2 FF&I budget -> is this because FF&I target is defined as having no methane, etc? sorry I am not familiar with the definition here, if it is UNFCCC/IPCC definition, maybe we can even link this here?